pkg/protecode/protecode.go

```diff
@@ -316,25 +316,36 @@ func (pc *Protecode) LoadReport(reportFileName string, productID int) *io.ReadCl
 }
 
 // UploadScanFile upload the scan file to the protecode server
-func (pc *Protecode) UploadScanFile(cleanupMode, group, filePath, fileName, version string, productID int, replaceBinary bool) *ResultData {
+func (pc *Protecode) UploadScanFile(cleanupMode, group, customDataJSONMap, filePath, fileName, version string, productID int, replaceBinary bool) *ResultData {
 	log.Entry().Debugf("[DEBUG] ===> UploadScanFile started.....")
 
 	deleteBinary := (cleanupMode == "binary" || cleanupMode == "complete")
 
 	var headers = make(map[string][]string)
+	if len(customDataJSONMap) > 0 {
+		customDataHeaders := map[string]string{}
+		if err := json.Unmarshal([]byte(customDataJSONMap), &customDataHeaders); err != nil {
+			log.Entry().Warn("[WARN] ===> customDataJSONMap flag must be a valid JSON map. Check the value of --customDataJSONMap and try again.")
+		} else {
+			for k, v := range customDataHeaders {
+				headers["META-"+strings.ToUpper(k)] = []string{v}
+			}
+		}
+	}
+
+	headers["Group"] = []string{group}
+	headers["Delete-Binary"] = []string{fmt.Sprintf("%v", deleteBinary)}
 
 	if (replaceBinary) && (version != "") {
 		log.Entry().Debugf("[DEBUG] ===> replaceBinary && version != empty ")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}, "Replace": {fmt.Sprintf("%v", productID)}, "Version": {version}}
+		headers["Replace"] = []string{fmt.Sprintf("%v", productID)}
+		headers["Version"] = []string{version}
 	} else if replaceBinary {
+		headers["Replace"] = []string{fmt.Sprintf("%v", productID)}
 		log.Entry().Debugf("[DEBUG] ===> replaceBinary")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}, "Replace": {fmt.Sprintf("%v", productID)}}
 	} else if version != "" {
 		log.Entry().Debugf("[DEBUG] ===> version != empty ")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}, "Version": {version}}
-	} else {
-		log.Entry().Debugf("[DEBUG] ===> replaceBinary is false and version == empty")
-		headers = map[string][]string{"Group": {group}, "Delete-Binary": {fmt.Sprintf("%v", deleteBinary)}}
+		headers["Version"] = []string{version}
```

Comment:
Here you can skip `header["Version"]`

Reply:
See my comment [here](https://github.com/SAP/jenkins-library/pull/3836/files#r989550248)